Tests/CartonTests/CartonTests.swift

```diff
@@ -1,7 +1,21 @@
 import class Foundation.Bundle
 import XCTest
+import CartonHelpers
+import TSCBasic
 
 final class CartonTests: XCTestCase {
+  /// Returns path to the built products directory.
+  var productsDirectory: URL {
+    #if os(macOS)
+    for bundle in Bundle.allBundles where bundle.bundlePath.hasSuffix(".xctest") {
+      return bundle.bundleURL.deletingLastPathComponent()
+    }
+    fatalError("couldn't find the products directory")
+    #else
+    return Bundle.main.bundleURL
+    #endif
+  }
+  
```

Comment:
**nitpick:** please apply `swiftformat .` or run `pre-commit install` to run it automatically before every commit to avoid these linter formatting warnings.